### PR TITLE
Fix issue with legend border

### PR DESCRIPTION
**Pull request by ghost**
Fixes issue #7766

### CODE REVIEW

**Comment by kodecount** [comment]
@duboisp @LaurentGoderre The form-style guide was updated. Please merge this when able to.

**Comment by duboisp** [comment]
@kodecount I will need to do some testing before to merge your PR. I think your PR remove completely the border-top where it should remain as the default behavior.

**Comment by kodecount** [comment]
DEFAULT VIEW:
default

WITH CLASS VIEW:
class

**Comment by kodecount** [comment]
DEFAULT VIEW:
default

The default view is not completely the same due to the border being present on each field set, but the original way it achieved the effect of only adding a border when there more then one field set was using a CSS hack. @cfarquharson and @LaurentGoderre Thoughts?

CLASS VIEW:
class

**Comment by kodecount** [comment]
@duboisp Fixed it to keep original code

**Review by duboisp — CHANGES_REQUESTED** [review]
Visually it's seem to work fine, but the fieldset keep his border-top where it is hidden because the float of the legend element is changed. I think the ```border-top``` style should be removed as the intention is to not have it.

Can you make adjustment where the class ```legend-brdr-bttm``` would be applied to the ```fieldset``` element instead of the ```legend``` element, then it would be easier to remove the border-top that was set on the fieldset.

**Comment by kodecount** [comment]
@duboisp It has been fixed to use the fieldset and not use the legend to create a border.

**Comment by kodecount** [comment]
DEFAULT VIEW (When their is more then one fieldset group):
default

CLASS VIEW (using the new `.legend-brdr-bttm` on a fieldset):
legend-brdr-bttm

**Comment by kodecount** [comment]
@duboisp Any ideas just did some testing works great in all but Firefox.

**Comment by kodecount** [comment]
@duboisp Firefox isn't working apparent their is a know issue. Have to use:
```
fieldset {
position: relative;
}
legend {
position: absolute;
}
```
or
```
legend {
float: left;
}
```
Their is another fix with `@-moz-document url-prefix()` but that breaks grunt.

**Comment by duboisp** [comment]
@kodecount 
> [...] but Firefox.

:confused:, it do not work as expected in Firefox??

May be it could be because the ```border-bottom``` should be set on the ```legend``` element instead of setting a ```border-top``` on the next elements following the ```legend```. 

FYI - As per the spec, the [```legend```](https://html.spec.whatwg.org/multipage/forms.html#the-legend-element) is the first chlid of a ```fieldset``` element. Also as the the [```fieldset```](https://html.spec.whatwg.org/multipage/forms.html#the-fieldset-element) spec, the ```legend``` is optional.

**Comment by kodecount** [comment]
@duboisp Firefox isn't working apparently their is a know issue. Have to use:
```
fieldset {
position: relative;
}
legend {
position: absolute;
}
```
or
```
legend {
float: left;
}
```
Their is another fix with `@-moz-document url-prefix()` which allows CSS just for Firefox, but that breaks grunt.

**Comment by kodecount** [comment]
@duboisp `ERROR: Unknown @ rule: @-moz-document. This rule looks for recoverable syntax errors. (errors) Browsers: All` this is the error with Firefox rules.

**Comment by kodecount** [comment]
@duboisp The new update should fix Firefox

**Comment by duboisp** [comment]
Thanks @kodecount, your solution seems to work fine, I tested on FF Dev. Chrome, Edge and IE11.

But I wonder the why the following css was used:

Why the top border was removed on the first child element of fieldset and not on the legend? [line 612](https://github.com/wet-boew/wet-boew/blob/master/src/base/bootstrap-overrides/_base.scss#L612)
```
&:first-child {
 		border-top: 0;
 	}
```
Why float was used in the legend? [line 601](https://github.com/wet-boew/wet-boew/blob/master/src/base/bootstrap-overrides/_base.scss#L601)
Why the bottom border was removed to the legend? [line 600](https://github.com/wet-boew/wet-boew/blob/master/src/base/bootstrap-overrides/_base.scss#L600)
```
legend {
	border-bottom: 0;
	float: left;
}
```

Then I just remarked this note:  [line 590](https://github.com/wet-boew/wet-boew/blob/master/src/base/bootstrap-overrides/_base.scss#L590)
```
/*
 *	Fieldset/legend border position
 *
 *	Bootstrap puts the border below the legend which creates a visual
 *	separation between the legend and the fields to which it relates. Putting
 *	the border above eliminates that visual separation. Putting it on the
 *	fieldset element instead ensures that all fieldsets will have a separation
 *	even when no legend is used.
 */
```

So, @LaurentGoderre what do you think of undoing then re-doing practice? Do you think it is ok for this use case?

**Comment by kodecount** [comment]
@duboisp The file you reference is not the code change.

**Comment by kodecount** [comment]
@duboisp The file you reference is not the code change. That is the original code.

What original code does:

The original code goes against bootstrap of having the border under the legend instead applies the border to above each `fieldset` if their is more then one `fieldset` the issue with this is Firefox behaves differently then most browsers and in order for the border to work the float is required. 

What my current code does:

What the latest version of the code change I provided does is apply the border to the `legend` and put the border above the `legend` or below the `legend` dependent on if a class is added to the `fieldset`. The downside to this method is their is no border when a legend is not added. I'm personally okay with this cause if your to separate fields their should be context to why.

**Comment by duboisp** [comment]
@kodecount Personally I am ok with your solution and as I said it works fine during my testing. 

But I just need a second opinion about the practice of undoing style then redoing style for this particular use case before to merge it.

File: src/base/bootstrap-overrides/_base.scss
Comment (LaurentGoderre) [inline src/base/bootstrap-overrides/_base.scss]: Please revrwite this in propper SCCSS

```scss
&:first-child {
    legend {
        [...]
    }
}
```

File: src/base/bootstrap-overrides/_base.scss
Comment (LaurentGoderre) [inline src/base/bootstrap-overrides/_base.scss]: Please rewrite this in proper SCSS

```scss
&.legend-brdr-bttm {
    legend {
        [...]
    }
}
```

**Review by LaurentGoderre — DISMISSED** [review]

**Comment by LaurentGoderre** [comment]
@duboisp I feel that it's risky and can create some conflicts in the future.

**Comment by kodecount** [comment]
@LaurentGoderre Newer bootstraps are actually removing borders all together.

**Comment by kodecount** [comment]
@LaurentGoderre Reverted code to use original method for top border.

**Comment by LaurentGoderre** [comment]
I like that it's class specific now so less risky

**Review by duboisp — APPROVED** [review]

**Comment by duboisp** [comment]
I tested, it work fine and it is [already documented](http://wet-boew.github.io/wet-boew-styleguide/v4/design/forms-en.html#fieldset). I am ok to merge this PR.
 
@LaurentGoderre do you have any other concern about this PR?

**Comment by LaurentGoderre** [comment]
No issue since it's isolated

**Comment by kodecount** [comment]
Don't forget to close #7766

**Pull request merged**
